lib/twingly/url/null_url.rb

```diff
@@ -24,6 +24,16 @@ def <=>(other)
         self.to_s <=> other.to_s
       end
 
+      def eql?(other)
+        return false unless other.is_a?(self.class)
```

Comment:
I wonder if it's possible to construct a spec that tests this. I tried commenting out the line, all specs still pass.

Reply:
(In an another way than introducing another test class that implements `hash` in the same way)

Reply:
Maybe we can do specific `eql?` testing?

```Diff
diff --git a/spec/lib/twingly/url/null_url_spec.rb b/spec/lib/twingly/url/null_url_spec.rb
index 09b9e04..6073698 100644
--- a/spec/lib/twingly/url/null_url_spec.rb
+++ b/spec/lib/twingly/url/null_url_spec.rb
@@ -112,6 +112,11 @@ describe Twingly::URL::NullURL do
       it "should be two unique objects" do
         expect([a,b].uniq).to eq([a,b])
       end
+
+      context ".eql?" do
+        subject { a.eql?(b) }
+        it { is_expected.to eq(false) }
+      end
     end
   end
 end
diff --git a/spec/lib/twingly/url_spec.rb b/spec/lib/twingly/url_spec.rb
index e8bb15c..f2f565f 100644
--- a/spec/lib/twingly/url_spec.rb
+++ b/spec/lib/twingly/url_spec.rb
@@ -884,6 +884,11 @@ describe Twingly::URL do
       it "should be two unique objects" do
         expect([a,b].uniq).to eq([a,b])
       end
+
+      context ".eql?" do
+        subject { a.eql?(b) }
+        it { is_expected.to eq(false) }
+      end
     end
   end

```

Reply:
good idea! switch the `context` to a `describe` though 😄